openedge-plugin/src/main/java/org/sonar/plugins/openedge/foundation/OpenEdgeSettings.java

```diff
@@ -210,16 +209,17 @@ private final void parseBuildDirectory() {
       if (f.getName().endsWith(".r")) {
         numRCode.incrementAndGet();
         service.submit(() -> {
-          try {
+          try (FileInputStream fis = new FileInputStream(f)) {
             LOG.debug("Parsing rcode {}", f.getAbsolutePath());
-            RCodeInfo rci = new RCodeInfo(new FileInputStream(f));
+            RCodeInfo rci = new RCodeInfo(fis);
             if (rci.isClass()) {
               numClasses.incrementAndGet();
-              numMethods.addAndGet(rci.getUnit().getMethods().size());
-              numProperties.addAndGet(rci.getUnit().getProperties().size());
+              numMethods.addAndGet(rci.getTypeInfo().getMethods().size());
+              numProperties.addAndGet(rci.getTypeInfo().getProperties().size());
+              proparseSession.injectTypeInfo(rci.getTypeInfo());
             }
-          } catch (InvalidRCodeException | IOException caught) {
-            LOG.error("Unable to parse rcode " + f.getAbsolutePath(), caught);
+          } catch (InvalidRCodeException | IOException | RuntimeException caught) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)